gkeys/gkeys/keyhandler.py

```diff
@@ -108,3 +111,27 @@ def key_search(self, args, first_match=False):
 
         self.logger.debug(_unicode("KeyHandler: key_search; keys = %s") % str(keys))
         return keys
+
+    def _is_expiring(self, results, days_limit):
+        is_expiring = False
+        for key in results:
+            for specs in results[key]:
+                if specs.days > days_limit*(-1) and specs.days < days_limit:
+                    is_expiring = True
+                    break
+        return is_expiring
+
+    def _generate_template(self, template_path, keyprints, specprint):
+        with open(template_path, 'r') as file_contents:
+            content = file_contents.read()
+        message_template = Template(content)
+        subkey_msg = ''
+        message = message_template.substitute(key_print=keyprints, spec_print=specprint)
+        return message
+
```

Comment:
this means you read in the template every time, for every key that you are generating an email for.  Seems a waste if it is the same template every time.  It would be better to have a set_template() then a generate_email() that simply filled in the contents with the keyprints.  Also this is a public method so no leading _